### DISCUSSION TITLE
Core dns responding with no data for valid DNS name

### DISCUSSION THREAD

**Discussion by nirvanagit** [coredns/coredns#6468] [General]
Hi,

We have aws lb dns going through core dns, and we observed during startup of our pods, (we use istio), the DNS resolution for valid entries return "no data".

Question:
1. Why would core dns return "no data" for a valid entry?
2. For this scenario, core dns did not log any errors, and all we saw was "NOERROR". Does coredns return NOERROR for "no data" ?

**Comment by chrisohaver** [comment]
You'll get a "no data" response if a record of the queried type does not exist, but one of another type with the same name does.  E.g. if a query is for an A record at `an.example.com.`, and there is no A record there, but there is a TXT record, or AAAA record, etc.

"no data" in DNS is shorthand for a response with an empty answer section, and a NOERROR response code.
This is a general DNS thing, not CoreDNS specific.
